File: official/docs/csharp/v6/smartrate/retrieve-standalone-smartrate-deliver-by.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using EasyPost;
using Newtonsoft.Json;

namespace EasyPostExamples
{
    public class Examples
    {
        public static async Task Main()
        {
            var client = new EasyPost.Client(new EasyPost.ClientConfiguration("EASYPOST_API_KEY"));

            EasyPost.Parameters.SmartRate.EstimateDeliveryDateForZipPair estimateDeliveryDateForZipPairParameters = new()
            {
                OriginPostalCode = address1Parameters.Zip,
                DestinationPostalCode = address2Parameters.Zip,
                PlannedShipDate = Fixtures.PlannedShipDate,
                Carriers = ["USPS"],
            };

            EasyPost.Models.API.EstimateDeliveryDateForZipPairResult results = await client.SmartRate.EstimateDeliveryDate(estimateDeliveryDateForZipPairParameters);

            Console.WriteLine(JsonConvert.SerializeObject(results, Formatting.Indented));
        }
    }
}

File: official/docs/java/v7/smartrate/retrieve-standalone-smartrate-deliver-by.java
package shipments;

import com.easypost.exception.EasyPostException;
import com.easypost.model.EstimateDeliveryDateForZipPairResult;
import com.easypost.service.EasyPostClient;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

public class RetrieveRecommendShipDate {
    public static void main(String[] args) throws EasyPostException {
        EasyPostClient client = new EasyPostClient("EASYPOST_API_KEY");
        Map<String, Object> params = new HashMap<String, Object>();
        params.put("from_zip", "10001");
        params.put("to_zip", "10002");
        params.put("planned_ship_date", "2024-07-18");
        params.put("carriers", Collections.singletonList("UPS"));
        
        EstimateDeliveryDateForZipPairResult results = client.smartRate
                .estimateDeliveryDate(params);

        System.out.println(results);
    }
}
